core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/PartitionInfo.scala

```diff
@@ -66,7 +79,7 @@ case class RangePartition(rangeColumnIndices: Seq[Int], rangeMin: Long, rangeMax
 /**
   * Represent the input stream is not partitioned and all data are on a single node.
   */
-case class SinglePartition() extends PartitionInfo {}
+final case class SinglePartition() extends PartitionInfo {}
```

Comment:
also change the new `BroadcastPartition` below to be `final`, for consistency

Reply:
done